crawl-ref/source/player-equip.cc

```diff
@@ -1281,9 +1281,22 @@ static void _equip_jewellery_effect(item_def &item, bool unmeld,
         break;
 
     case AMU_THE_GOURMAND:
-        // What's this supposed to achieve? (jpeg)
-        you.duration[DUR_GOURMAND] = 0;
-        mpr("You feel a craving for the dungeon's cuisine.");
+        if (you.species == SP_SPRIGGAN 
```

Comment:
this conditional works, but can it be replaced with more abstracted functions of some kind, rather than direct species checks? Spriggans already have MUT_HERBIVOROUS inherently, and the other two are undead. For the next conditional block all the species should inherently have MUT_CARNIVOROUS. That way this function wouldn't need to be edited when new species with these properties are added.